cloudinit/sources/helpers/azure.py

```diff
@@ -214,26 +214,29 @@ def report_compressed_event(event_name, event_content):
 def push_log_to_kvp(file_name=CFG_BUILTIN['def_log_file']):
     """Push a portion of cloud-init.log file or the whole file to KVP
     based on the file size.
-    If called more than once, it skips pushing the log file to KVP again."""
+    If called more than once, it continues from where it left off."""
 
-    log_pushed_to_kvp = bool(os.path.isfile(LOG_PUSHED_TO_KVP_MARKER_FILE))
-    if log_pushed_to_kvp:
-        report_diagnostic_event(
-            "cloud-init.log is already pushed to KVP", logger_func=LOG.debug)
-        return
+    last_log_byte_pushed_to_kvp_index = 0
+    try:
+        with open(LOG_PUSHED_TO_KVP_INDEX_FILE, "r") as f:
+            last_log_byte_pushed_to_kvp_index = int(f.read())
+    except Exception as ex:
+        report_diagnostic_event("Failed to get the last log byte pushed to KVP"
+                                ": %s." % repr(ex), logger_func=LOG.debug)
 
     LOG.debug("Dumping cloud-init.log file to KVP")
     try:
         with open(file_name, "rb") as f:
             f.seek(0, os.SEEK_END)
-            seek_index = max(f.tell() - MAX_LOG_TO_KVP_LENGTH, 0)
+            seek_index = max(f.tell() - MAX_LOG_TO_KVP_LENGTH,
+                             last_log_byte_pushed_to_kvp_index)
             report_diagnostic_event(
```

Comment:
Report the value of `last_log_byte_pushed_to_kvp_index` so that we know the index being pushed.